CRUD added for Treatment + Medicine (not complete)

anamariaghiorghiu opened the pull request.

anamariaghiorghiu committed:
Fixed some Unit Tests

anamariaghiorghiu committed:
Merge branch 'dev' into dev-mariuca

anamariaghiorghiu committed:
treatments

anamariaghiorghiu committed:
integration test fail

Al-Andrew committed:
Fixed integration tests

anamariaghiorghiu committed:
ui + some integration tests

anamariaghiorghiu committed:
Medicine almost done + changes for Treatment

OpariucRares commented on src/DucksNet.API/Controllers/TreatmentController.cs:
```diff
@@ -9,9 +11,11 @@ namespace DucksNet.API.Controllers;
 {
     private readonly IRepository<Treatment> _treatmentRepository;
-    public TreatmentController(IRepository<Treatment> repository)
+    //private readonly IRepository<Medicine> _medicineRepository;
```
mark as a TODO (abbreviation of your name)

OpariucRares commented on src/DucksNet.Domain/Model/Medicine.cs:
```diff
@@ -1,5 +1,6 @@
 ﻿using DucksNet.SharedKernel.Utils;
 using DucksNet.Domain.Model.Enums;
+using System.Net;
```
does it need?

OpariucRares commented on tests/DucksNet.IntegrationTests/TreatmentControllerTests.cs:
```diff
@@ -0,0 +1,10 @@
+//using System.Text;
+//using System.Threading.Tasks;
+
+//namespace DucksNet.IntegrationTests;
```
why is it commented? make at least one test

OpariucRares commented on tests/DucksNet.UnitTests/TreatmentTests.cs:
```diff
@@ -1,18 +1,34 @@
+    //    result.Value.Should().BeNull();
+    //    result.Errors.Should().NotBeEmpty();
+    //    result.Errors.Should().Contain("Can't create treatment with no medicine.");
+    //}
```
same here. At least make the treatment domain 100%

OpariucRares reviewed (changes requested):
don't add the DucksNet.db, DucksNet.db-shm, DucksNet.db-wal. remove for the push, but keet it local

OpariucRares committed:
Merge branch 'dev' into dev-mariuca

# Conflicts:
#	tests/DucksNet.IntegrationTests/MedicalRecordsControllerTests.cs

OpariucRares committed:
Merge branch 'dev' into dev-mariuca

OpariucRares committed:
removed by mistake the url

anamariaghiorghiu committed:
help

anamariaghiorghiu committed:
mesaj
Merge branch 'dev-mariuca' of https://github.com/Ducks-Net/SiDiez into dev-mariuca

anamariaghiorghiu committed:
CRUD added for Treatment + Medicine

OpariucRares committed:
Merge branch 'dev' into dev-mariuca

# Conflicts:
#	src/DucksNet.WebUI/Program.cs
#	src/DucksNet.WebUI/Shared/NavMenu.razor
#	tests/DucksNet.IntegrationTests/MedicalRecordsControllerTests.cs

OpariucRares committed:
removed useless files

OpariucRares committed:
yes

OpariucRares committed:
added treatment integration tests

OpariucRares committed:
codefactor bs

OpariucRares commented on src/DucksNet.Domain/Model/Medicine.cs:
```diff
@@ -21,14 +26,63 @@ private Medicine(string name, string description, double price, DrugAdministrati
+        {
+            Name = name;
+        }
+        else if (string.IsNullOrWhiteSpace(name))
```
you can remove the ifs; let alone else

OpariucRares reviewed (changes requested):
make for all the ifs

Al-Andrew commented on src/DucksNet.Domain/Model/Medicine.cs:
```diff
@@ -21,14 +26,63 @@ private Medicine(string name, string description, double price, DrugAdministrati
+        {
+            return Result.Error("Description is empty.");
+        }
+        if (price >= 0)
+        {
+            Price = price;
+        }
+        else if (price <= 0)
+        {
+            return Result.Error("Invalid price.");
+        }
```
the if and elif overlap on price = 0 here

Al-Andrew commented on tests/DucksNet.IntegrationTests/BaseIntegrationTests.cs:
```diff
@@ -34,7 +34,7 @@ protected void ClearDatabase()
         context.RemoveRange(context.MedicalRecords);
         context.RemoveRange(context.Employees);
-        context.RemoveRange(context.Treatments);
+        //context.RemoveRange(context.Treatments);
```
Why should we not clean out the Treatments?

Al-Andrew commented on tests/DucksNet.IntegrationTests/MedicineControllerTests.cs:
```diff
@@ -0,0 +1,158 @@
+        getMedicineResult.Content.Headers.ContentLength.Should().Be(0);
+    }
+
+    //[Fact]
+    //public async Task When_CreatedMedicineWithInvalidPrice_Then_ShouldReturnFail()
+    //{
+    //    ClearDatabase();
+    //    //Arrange
+    //    Guid id = Guid.NewGuid();
+    //    Medicine medicine = new Medicine(id, "Naldorex" , "Migraine", -999, DrugAdministration.Oral);
+    //    //Act
+    //    var medicineResponse = await TestingClient.PostAsJsonAsync(MedicineUrl, medicine);
+    //    //Assert
+    //    medicineResponse.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+    //}
```
Why is this test commented out?

Al-Andrew reviewed (approved):
All in all seems fine. But please uncomment out the removing of treatments fro the database when testing.

OpariucRares committed:
Merge branch 'dev' into dev-mariuca

OpariucRares committed:
modified from review

OpariucRares committed:
Merge branch 'dev' into dev-mariuca
